### PR TITLE
Update HotShot to tag 0.4.14

### LINKED ISSUE #598: Update to latest HotShot for Cortado deployment
HotShot has working catchup now, so we definitely want to pull that into Cortado if possible

### PR DESCRIPTION
**Pull request by nomaxg**
Depends on: https://github.com/EspressoSystems/hotshot-query-service/pull/221

Closes #598

### CODE REVIEW

File: .github/workflows/build_static.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ on:
   pull_request:
   workflow_dispatch:
 
+env:
+  RUSTFLAGS: "--cfg async_executor_impl=\"async-std\" --cfg async_channel_impl=\"async-std\""
```
Comment (nomaxg) [inline .github/workflows/build_static.yml:16]: @elliedavidson @rob-maron would it be possible to have HotShot build with a default async runtime?

File: sequencer/src/lib.rs [hunk 2]
```diff
@@ -224,7 +222,7 @@ pub type Storage = MemoryStorage<SeqTypes, Leaf>;
 pub type Event = hotshot::types::Event<SeqTypes, Leaf>;
 
 pub type SignatureSchemeType = BLSSignatureScheme;
-pub type SignatureKeyType = JfPubKey<SignatureSchemeType>;
+pub type SignatureKeyType = BN254Pub;
```
Comment (sveitser) [inline sequencer/src/lib.rs:225]: Is `SignatureSchemeType` unused now?

File: .github/workflows/build_static.yml (hunk 1 above)
Comment (rob-maron) [reply .github/workflows/build_static.yml:16]: We could probably change some of the conditional compilation attributes to use `async-std` if none are specified. Would that be an alright solution? Other than that, I don't think it's possible to set default `cfg` values without editing something like `.cargo/config.toml`.

Comment (nomaxg) [reply .github/workflows/build_static.yml:16]: yeah that sounds good!

**Review by sveitser — APPROVED** [review]

File: sequencer/src/lib.rs (hunk 2 above)
Comment (nomaxg) [reply sequencer/src/lib.rs:225]: yeah looks like it

File: .github/workflows/build_static.yml (hunk 1 above)
Comment (elliedavidson) [reply .github/workflows/build_static.yml:16]: @DieracDelta Can you do this update when you update the profiles in our justfile? :)

**Pull request merged**